src/System.Net.Requests/tests/HttpWebResponseTest.cs

```diff
@@ -12,6 +14,158 @@ namespace System.Net.Tests
+                // ANSI
+                "ddd MMM d HH:mm:ss yyyy",
+            };
+
+            var offset_formats = new[]
+            {
+                // RFC1123 - Offset
+                "ddd, dd MMM yyyy HH:mm:ss zzz",
+                // RFC850 - Offset
+                "dddd, dd-MMM-yy HH:mm:ss zzz",
+            };
+
+            var dates = new[]
+            {
+                new DateTimeOffset(2018, 1, 1, 12, 1, 14, TimeSpan.Zero),
+                new DateTimeOffset(2018, 1, 3, 15, 0, 0, TimeSpan.Zero),
+                new DateTimeOffset(2015, 5, 6, 20, 45, 38, TimeSpan.Zero),
+            };
+
+            foreach (var date in dates)
+            {
+                var expected = date.LocalDateTime;
+
+                foreach (var format in zero_formats.Concat(offset_formats))
+                {
+                    var formatted = date.ToString(format);
+                    yield return new object[] { formatted, expected };
+                }
+            }
+
+            foreach (var format in offset_formats)
+            {
+                foreach (var date in dates.SelectMany(d => new[] { d.ToOffset(TimeSpan.FromHours(5)), d.ToOffset(TimeSpan.FromHours(-5)) }))
+                {
+                    var formatted = date.ToString(format);
+                    var expected = date.LocalDateTime;
+                    yield return new object[] { formatted, expected };
+                    yield return new object[] { formatted.ToLowerInvariant(), expected };
+                }
+            }
+        }
+
+        public static IEnumerable<object[]> Dates_Invalid_Data()
+        {
+            yield return new object[] { "not a valid date here" };
+            yield return new object[] { "Sun, 31 Nov 1234567890 33:77:80 GMT" };
+
+            // Should be Sunday...
+            yield return new object[] { "Sat, 25 Mar 2018 16:33:01 GMT" };
+            yield return new object[] { "Sat, 25 Mar 2018 16:33:01 UTC" };
+            yield return new object[] { "Sat, 25 Mar 2018 21:33:01+05:00" };
+            yield return new object[] { "Saturday, 25-Mar-18 16:33:01 GMT" };
+            yield return new object[] { "Saturday, 25-Mar-18 16:33:01 UTC" };
+            yield return new object[] { "Saturday, 25-Mar-18 21:33:01+05:00" };
+            yield return new object[] { "Sat Mar 25 21:33:01 2018" };
+            // Invalid day-of-week/month values
+            yield return new object[] { "Sue, 25 Not 2018 16:33:01 GMT" };
+            yield return new object[] { "Sue, 25 Not 2018 16:33:01 UTC" };
+            yield return new object[] { "Sue, 25 Not 2018 21:33:01+05:00" };
+            yield return new object[] { "Surprise, 25-Not-18 16:33:01 GMT" };
+            yield return new object[] { "Surprise, 25-Not-18 16:33:01 UTC" };
+            yield return new object[] { "Surprise, 25-Not-18 21:33:01+05:00" };
+            yield return new object[] { "Sue Not 25 21:33:01 2018" };
+            // Strange separators
```

Comment:
Similar to above - I don't think it makes sense to have more than 1 invalid separator in each test case. In theory the first one should trigger the error.